### ISSUE TITLE
Invalid block name "P1 ro" in ELF file compiled with IAR

### ISSUE THREAD

**Issue by autinerd** [NationalSecurityAgency/ghidra#2520]
**Describe the bug**
The IAR C compiler/linker uses Section names like "P1 ro", "P2 rw", "P2 zi" etc. to indicate read-only, read-write, zero-initialized etc. code.

The import of an ELF file resolves in an exception:

```
Invalid block name: P1 ro
java.lang.IllegalArgumentException: Invalid block name: P1 ro
	at ghidra.program.database.mem.MemoryMapDB.checkBlockName(MemoryMapDB.java:821)
	at ghidra.program.database.mem.MemoryMapDB.createInitializedBlock(MemoryMapDB.java:646)
	at ghidra.app.util.MemoryBlockUtils.createInitializedBlock(MemoryBlockUtils.java:227)
	at ghidra.app.util.opinion.ElfProgramBuilder.createInitializedBlock(ElfProgramBuilder.java:3035)
	at ghidra.app.util.opinion.MemorySectionResolver.processSectionRanges(MemorySectionResolver.java:372)
	at ghidra.app.util.opinion.MemorySectionResolver.resolveSectionMemory(MemorySectionResolver.java:284)
	at ghidra.app.util.opinion.MemorySectionResolver.resolve(MemorySectionResolver.java:261)
	at ghidra.app.util.opinion.ElfProgramBuilder.load(ElfProgramBuilder.java:130)
	at ghidra.app.util.opinion.ElfProgramBuilder.loadElf(ElfProgramBuilder.java:100)
	at ghidra.app.util.opinion.ElfLoader.load(ElfLoader.java:153)
	at ghidra.app.util.opinion.AbstractLibrarySupportLoader.doLoad(AbstractLibrarySupportLoader.java:347)
	at ghidra.app.util.opinion.AbstractLibrarySupportLoader.loadProgram(AbstractLibrarySupportLoader.java:83)
	at ghidra.app.util.opinion.AbstractProgramLoader.load(AbstractProgramLoader.java:112)
	at ghidra.plugin.importer.ImporterUtilities.importSingleFile(ImporterUtilities.java:400)
	at ghidra.plugin.importer.ImporterDialog.lambda$okCallback$7(ImporterDialog.java:349)
	at ghidra.util.task.TaskLauncher$1.run(TaskLauncher.java:90)
	at ghidra.util.task.Task.monitoredRun(Task.java:124)
	at ghidra.util.task.TaskRunner.lambda$startTaskThread$1(TaskRunner.java:94)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)

---------------------------------------------------
Build Date: 2020-Nov-13 1112 EST
Ghidra Version: 9.2
JVM Version: Oracle Corporation 14.0.2
OS: Windows 10 10.0 amd64
```

In this function are block names with spaces rejected:

https://github.com/NationalSecurityAgency/ghidra/blob/14d4c87ef43f45e2f50edfa99c1e4b7f5779e232/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/mem/Memory.java#L823-L834

A current workaround is to replace the spaces with an underscore etc. in a hex editor before importing to Ghidra, but this is quite inconvenient.

**Possible solutions:**

- Ask the user at import if unsupported characters should be replaced with an underscore
- Just replace it (print a warning that a replacement happened)

**Comment by ghizard** [comment]
Also see #2465

**Comment by autinerd** [comment]
@ghizard Oh thanks, I had searched for this, but didn't found anything. But good to know!

**Comment by ghidra1** [comment]
This has been fixed for 9.2.1 and has been merged into patch and master branches
